R/mle-finder.R

```diff
@@ -21,3 +21,76 @@ linear_BFGS <- function(design, outcome, noise_var = 1) {
   return(MLE)
 }
 
+logit_loglik <- function(coef, design, outcome) {
+
+  n <- length(outcome)
+  log_lik <- sum(sapply(1:n, function(i){
+    outcome[i] * sum(design[i,] * coef) -
+      log(1 + exp(sum(design[i,] * coef)))
+    }))
+  return(log_lik)
+
+}
+
+logit_gradient <- function(coef, design, outcome) {
+
+  pi <- exp(design %*% coef) / (1 + exp(design %*% coef))
+  gradient <- as.vector(t(outcome - pi) %*% design)
+
+  return(gradient)
+
+}
+
+logit_BFGS <- function(design, outcome) {
```

Comment:
-2

This function is almost identical to your linear regression version. Better to write a single `bfgs()` function that takes in design, outcome, function, and gradient. Then you can pass in the desired likelikhood and gradient. Additional arguments like `noise_var` from the lm can be handled with the `...` arguments.